registry/handlers/blobupload.go

```diff
@@ -97,6 +97,11 @@ func (buh *blobUploadHandler) StartBlobUpload(w http.ResponseWriter, r *http.Req
 	}
 
 	w.Header().Set("Docker-Upload-UUID", buh.Upload.ID())
+
+	if r.FormValue("digest") != "" {
+		w.WriteHeader(http.StatusCreated)
```

Comment:
Huh...I just realized that I don't understand why the `w.WriteHeader(http.StatusAccepted)` below doesn't overwrite this. But also it occurs to me that we should probably also check for non-empty request body.

Reply:
This change looks good to me

Reply:
`WriteHeader` (at least theoretically) starts sending an HTTP response to the client, so once you've called it you can't change your mind about what status code to send. Thus, the `ResponseWriter` keeps track of whether it's already been called and ignores any subsequent calls.

Still best to only call it once, though :+1: